dip-template/runtimes/dip-consumer/src/origin_adapter.rs

```diff
@@ -0,0 +1,57 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate::{AccountId, DidIdentifier, MerkleProofVerifierOutput, RuntimeOrigin, Web3Name};
+use frame_support::traits::EnsureOrigin;
+use pallet_dip_consumer::{DipOrigin, EnsureDipOrigin};
+use pallet_postit::traits::Usernamable;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_core::RuntimeDebug;
+
+pub struct EnsureDipOriginAdapter;
+
+impl EnsureOrigin<RuntimeOrigin> for EnsureDipOriginAdapter {
+	type Success = DipOriginAdapter;
+
+	fn try_origin(o: RuntimeOrigin) -> Result<Self::Success, RuntimeOrigin> {
+		EnsureDipOrigin::try_origin(o).map(DipOriginAdapter)
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_successful_origin() -> Result<RuntimeOrigin, ()> {
+		// TODO: Replace with actual DIP origin upon benchmarking
+		Ok(RuntimeOrigin::root())
+	}
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct DipOriginAdapter(DipOrigin<DidIdentifier, AccountId, MerkleProofVerifierOutput>);
+
+impl Usernamable for DipOriginAdapter {
```

Comment:
`DipOrigin` is generic over what details are included in it: 

```rust
pub struct DipOrigin<Identifier, AccountId, Details> {
	/// The subject identifier which is performing the DIP operation.
	pub identifier: Identifier,
	/// The local account address of the tx submitter.
	pub account_address: AccountId,
	/// Details returned by the proof verifier upon successful proof
	/// verification.
	pub details: Details,
}
```

The trait could be implemented on `DipOrigin<...> where Details: GetUsername`, but this would mean the `post_it` pallet now depends on the `dip_consumer` pallet. As the PostIt pallet is meant to be shown as an example, and since different adapters might try to "extrapolate" different information from a given DIP origin, I think it would make sense to keep it in this way.